scylla/src/transport/load_balancing/default.rs

```diff
@@ -1575,7 +1601,8 @@ mod tests {
                 expected_groups: ExpectedGroupsBuilder::new()
                     .deterministic([B]) // pick local rack replicas
                     .deterministic([C]) // fallback replicas
-                    .group([A, G]) // local nodes
+                    .group([A]) // local rack nodes
```

Comment:
> Considering non-token-aware queries (e.g. unprepared statements), they would never prioritize the preferred rack.

I'd like to see a test which has `routing_info` with a `token` field set to `None`.